src/diff_edit.rs

```diff
@@ -240,21 +246,32 @@ pub fn run_mergetool(
     }
 
     let output_file_contents: Vec<u8> = std::fs::read(paths.get("output").unwrap())?;
-    if output_file_contents.is_empty() || output_file_contents == materialized_conflict {
+    if output_file_contents.is_empty() || output_file_contents == initial_output_content {
         return Err(ConflictResolveError::EmptyOrUnchanged);
     }
-    // TODO: parse any remaining conflicts (done in followup commit)
-    let new_file_id = tree
-        .store()
-        .write_file(repo_path, &mut File::open(paths.get("output").unwrap())?)?;
-    let mut tree_builder = tree.store().tree_builder(tree.id().clone());
-    tree_builder.set(
-        repo_path.clone(),
+
+    let mut new_tree_value: Option<TreeValue> = None;
```

Comment:
nit: you probably don't need this type annotation (but feel free to leave it if you prefer to have it anyway)

Reply:
I like it here for readablity.